backend/pom.xml

```diff
@@ -79,76 +72,50 @@
                     <header>${project.basedir}/GPL-3.txt</header>
                 </configuration>
             </plugin>
+
+
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
-                <version>2.3.1</version>
+                <version>2.4</version>
                 <configuration>
                     <archive>
                         <manifest>
                             <addClasspath>true</addClasspath>
                             <mainClass>cz.cuni.mff.xrg.odcs.backend.AppEntry</mainClass>
                             <classpathPrefix>lib/</classpathPrefix>
+                            <classpathLayoutType>custom</classpathLayoutType>
+                            <customClasspathLayout>$${artifact.artifactId}-$${artifact.version}$${dashClassifier?}-$${artifact.extension}/</customClasspathLayout>
```

Comment:
When extracting dependencies into the lib directory, resource clashes can occur, i.e. if two dependencies pull in identically named files. Therefore each dependency is extracted into a subdirectory of lib. Therefore the classpath needs to be adjusted so that classes can be found in each of these dependency subdirectories.